### ISSUE TITLE
v5.21.7-25-g7143f21 breaks REHSACK/List-MoreUtils-0.401.tar.gz

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#14355]
Migrated from [rt.perl.org#123479](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=123479) (status was 'resolved')

Searchable as RT123479$

**Comment by p5pRT** [comment]
### From @andk

bisect


---
commit 7143f21c482af816521c0e2affd8270cb71e7f9a
Author&#8203;: Father Chrysostomos \<sprout@&#8203;cpan\.org>
Date&#8203;:   Sat Dec 20 07&#8203;:14&#8203;:01 2014 \-0800

&nbsp;    Disallow GIMME in ext/

diagnostics


---
At the time of this writing there are no fail reports yet on
cpantesters\, so I copy the diagnostics here&#8203;:

cc \-c  \-I\. \-fwrapv \-fno\-strict\-aliasing \-pipe \-fstack\-protector\-strong \-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64 \-D\_FORTIFY\_SOURCE=2 \-O2 \-g   \-DVERSION=\\"0\.401\\" \-DXS\_VERSION=\\"0\.401\\" \-fPIC "\-I/home/sand/src/perl/repoperls/installed\-perls/perl/v5\.21\.7\-44\-gef3a1d8/127e/lib/5\.21\.8/x86\_64\-linux\-ld/CORE"  \-DPERL\_EXT MoreUtils\.c
MoreUtils\.xs&#8203;: In function ‘XS\_List\_\_MoreUtils\_uniq’&#8203;:
MoreUtils\.xs&#8203;:1570&#8203;:6&#8203;: error&#8203;: ‘GIMME’ undeclared \(first use in this function\)
&nbsp;  if \(GIMME == G\_SCALAR\) \{
&nbsp;      ^
MoreUtils\.xs&#8203;:1570&#8203;:6&#8203;: note&#8203;: each undeclared identifier is reported only once for each function it appears in
MoreUtils\.xs&#8203;: In function ‘XS\_List\_\_MoreUtils\_bsearch’&#8203;:
MoreUtils\.xs&#8203;:1792&#8203;:17&#8203;: error&#8203;: ‘GIMME’ undeclared \(first use in this function\)
&nbsp;     I32 gimme = GIMME; /\* perl\-5\.5\.4 bus\-errors out later when using GIMME
&nbsp;                 ^
Makefile&#8203;:354&#8203;: recipe for target 'MoreUtils\.o' failed
make&#8203;: \*\*\* \[MoreUtils\.o\] Error 1
\(/usr/bin/make exited with 512\)


perl \-V


---
Summary of my perl5 \(revision 5 version 21 subversion 8\) configuration&#8203;:
&nbsp;  Commit id&#8203;: 7143f21c482af816521c0e2affd8270cb71e7f9a
&nbsp;  Platform&#8203;:
&nbsp;    osname=linux\, osvers=3\.16\.0\-4\-amd64\, archname=x86\_64\-linux\-thread\-multi
&nbsp;    uname='linux k83 3\.16\.0\-4\-amd64 \#1 smp debian 3\.16\.7\-2 \(2014\-11\-06\) x86\_64 gnulinux '
&nbsp;    config\_args='\-Dprefix=/home/sand/src/perl/repoperls/installed\-perls/perl/v5\.21\.7\-25\-g7143f21/9980 \-Dmyhostname=k83 \-Dinstallusrbinperl=n \-Uversiononly \-Dusedevel \-des \-Ui\_db \-Duseithreads \-Uuselongdouble \-DDEBUGGING=\-g'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    use64bitint=define\, use64bitall=define\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-D\_REENTRANT \-D\_GNU\_SOURCE \-fwrapv \-fno\-strict\-aliasing \-pipe \-fstack\-protector\-strong \-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64 \-D\_FORTIFY\_SOURCE=2'\,
&nbsp;    optimize='\-O2 \-g'\,
&nbsp;    cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-fwrapv \-fno\-strict\-aliasing \-pipe \-fstack\-protector\-strong \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='4\.9\.2'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=12345678\, doublekind=3
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16\, longdblkind=3
&nbsp;    ivtype='long'\, ivsize=8\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='cc'\, ldflags =' \-fstack\-protector\-strong \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib/gcc/x86\_64\-linux\-gnu/4\.9/include\-fixed /usr/include/x86\_64\-linux\-gnu /usr/lib /lib/x86\_64\-linux\-gnu /lib/\.\./lib /usr/lib/x86\_64\-linux\-gnu /usr/lib/\.\./lib /lib
&nbsp;    libs=\-lnsl \-lgdbm \-ldb \-ldl \-lm \-lcrypt \-lutil \-lc \-lpthread \-lgdbm\_compat
&nbsp;    perllibs=\-lnsl \-ldl \-lm \-lcrypt \-lutil \-lc \-lpthread
&nbsp;    libc=libc\-2\.19\.so\, so=so\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version='2\.19'
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E'
&nbsp;    cccdlflags='\-fPIC'\, lddlflags='\-shared \-O2 \-g \-L/usr/local/lib \-fstack\-protector\-strong'


Characteristics of this binary \(from libperl\)&#8203;: 
&nbsp;  Compile\-time options&#8203;: HAS\_TIMES MULTIPLICITY PERLIO\_LAYERS
&nbsp;                        PERL\_DONT\_CREATE\_GVSV
&nbsp;                        PERL\_HASH\_FUNC\_ONE\_AT\_A\_TIME\_HARD
&nbsp;                        PERL\_IMPLICIT\_CONTEXT PERL\_MALLOC\_WRAP
&nbsp;                        PERL\_NEW\_COPY\_ON\_WRITE PERL\_PRESERVE\_IVUV
&nbsp;                        PERL\_USE\_DEVEL USE\_64\_BIT\_ALL USE\_64\_BIT\_INT
&nbsp;                        USE\_ITHREADS USE\_LARGE\_FILES USE\_LOCALE
&nbsp;                        USE\_LOCALE\_COLLATE USE\_LOCALE\_CTYPE
&nbsp;                        USE\_LOCALE\_NUMERIC USE\_LOCALE\_TIME USE\_PERLIO
&nbsp;                        USE\_PERL\_ATOF USE\_REENTRANT\_API
&nbsp;  Built under linux
&nbsp;  Compiled at Dec 22 2014 19&#8203;:22&#8203;:05
&nbsp;  @&#8203;INC&#8203;:
&nbsp;    /home/sand/src/perl/repoperls/installed\-perls/perl/v5\.21\.7\-25\-g7143f21/9980/lib/site\_perl/5\.21\.8/x86\_64\-linux\-thread\-multi
&nbsp;    /home/sand/src/perl/repoperls/installed\-perls/perl/v5\.21\.7\-25\-g7143f21/9980/lib/site\_perl/5\.21\.8
&nbsp;    /home/sand/src/perl/repoperls/installed\-perls/perl/v5\.21\.7\-25\-g7143f21/9980/lib/5\.21\.8/x86\_64\-linux\-thread\-multi
&nbsp;    /home/sand/src/perl/repoperls/installed\-perls/perl/v5\.21\.7\-25\-g7143f21/9980/lib/5\.21\.8
&nbsp;    \.

\-\- 
andreas

**Comment by p5pRT** [comment]
### From @mauke

Am 22\.12\.2014 um 19&#8203;:36 schrieb \(Andreas J\. Koenig\) \(via RT\)&#8203;:
> \# New Ticket Created by  \(Andreas J\. Koenig\)
> \# Please include the string&#8203;:  \[perl \#123479\]
> \# in the subject line of all future correspondence about this issue\.
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=123479 >
>
>
> bisect
> \-\-\-\-\-\-
> commit 7143f21c482af816521c0e2affd8270cb71e7f9a
> Author&#8203;: Father Chrysostomos \<sprout@&#8203;cpan\.org>
> Date&#8203;:   Sat Dec 20 07&#8203;:14&#8203;:01 2014 \-0800
>
>      Disallow GIMME in ext/

See also https://rt.cpan.org/Public/Bug/Display.html?id=101067

\-\- 
Lukas Mai \<plokinom@&#8203;gmail\.com>

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @cpansprout

This was fixed in List&#8203;::MoreUtils 0\.402\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
